packages/astro/src/transitions/router.ts

```diff
@@ -13,9 +13,12 @@ type Events = 'astro:page-load' | 'astro:after-swap';
 // only update history entries that are managed by us
 // leave other entries alone and do not accidently add state.
 const persistState = (state: State) => history.state && history.replaceState(state, '');
-export const supportsViewTransitions = !!document.startViewTransition;
+
+export const supportsViewTransitions = import.meta.env.SSR === false && !!document.startViewTransition;
+
 export const transitionEnabledOnThisPage = () =>
-	!!document.querySelector('[name="astro-view-transitions-enabled"]');
+	import.meta.env.SSR === false && !!document.querySelector('[name="astro-view-transitions-enabled"]');
```

Comment:
Can't we use `supportsViewTransitions && !!document...` here instead?

Reply:
I went with import.meta because it has implications on tree-shaking, but it might extend to variables as well. I'll see.

Reply:
Like ematipico's idea. After `supportsViewTransitions` is once defined using `import.meta.env.SSR` we should be good without further `import.meta.env.SSR` expression and it looks less brutal :)